### PR TITLE
Fix: calling AWS Bedrock models

### LINKED ISSUE #9812: [Bug]: While creating instances of bedrock using sagemaker notebook, region parameter is not working
### Bug Description

I'm initiating bedrock titan LLM model using sagemaker notebook instance which is in Middle East region. Since bedrock is not supported in Middle East so while initiating I'm passing aws_region_name param as 'us-east-1' but it is still taking middle east region and giving error.

```
from llama_index.llms import Bedrock

llm = Bedrock(model="amazon.titan-text-express-v1",aws_region_name="us-east-1")
```

When I try to take inference from the llm initiated above, it gives following error:

> EndpointConnectionError: Could not connect to the endpoint URL: "https://bedrock-runtime.me-central-1.amazonaws.com/model/amazon.titan-text-express-v1/invoke"

Instead of using the region provided in aws_region_name param it is using region where my notebook is hosted.

### Version

0.9.24

### Steps to Reproduce

Create a notebook on some region may be in Middle East and while initiating bedrock pass some other region

```
from llama_index.llms import Bedrock

llm = Bedrock(model="amazon.titan-text-express-v1",aws_region_name="us-east-1")
```

use llm to generate text or chat.

### Relevant Logs/Tracbacks

```shell
gaierror                                  Traceback (most recent call last)
File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connection.py:174, in HTTPConnection._new_conn(self)
    173 try:
--> 174     conn = connection.create_connection(
    175         (self._dns_host, self.port), self.timeout, **extra_kw
    176     )
    178 except SocketTimeout:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/util/connection.py:72, in create_connection(address, timeout, source_address, socket_options)
     68     return six.raise_from(
     69         LocationParseError(u"'%s', label empty or too long" % host), None
     70     )
---> 72 for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
     73     af, socktype, proto, canonname, sa = res

File ~/anaconda3/envs/python3/lib/python3.10/socket.py:955, in getaddrinfo(host, port, family, type, proto, flags)
    954 addrlist = []
--> 955 for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
    956     af, socktype, proto, canonname, sa = res

gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

NewConnectionError                        Traceback (most recent call last)
File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/httpsession.py:464, in URLLib3Session.send(self, request)
    463 request_target = self._get_request_target(request.url, proxy_url)
--> 464 urllib_response = conn.urlopen(
    465     method=request.method,
    466     url=request_target,
    467     body=request.body,
    468     headers=request.headers,
    469     retries=Retry(False),
    470     assert_same_host=False,
    471     preload_content=False,
    472     decode_content=False,
    473     chunked=self._chunked(request.headers),
    474 )
    476 http_response = botocore.awsrequest.AWSResponse(
    477     request.url,
    478     urllib_response.status,
    479     urllib_response.headers,
    480     urllib_response,
    481 )

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connectionpool.py:799, in HTTPConnectionPool.urlopen(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)
    797     e = ProtocolError("Connection aborted.", e)
--> 799 retries = retries.increment(
    800     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
    801 )
    802 retries.sleep()

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/util/retry.py:525, in Retry.increment(self, method, url, response, error, _pool, _stacktrace)
    523 if self.total is False and error:
    524     # Disabled, indicate to re-raise the error.
--> 525     raise six.reraise(type(error), error, _stacktrace)
    527 total = self.total

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/packages/six.py:770, in reraise(tp, value, tb)
    769         raise value.with_traceback(tb)
--> 770     raise value
    771 finally:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connectionpool.py:715, in HTTPConnectionPool.urlopen(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)
    714 # Make the request on the httplib connection object.
--> 715 httplib_response = self._make_request(
    716     conn,
    717     method,
    718     url,
    719     timeout=timeout_obj,
    720     body=body,
    721     headers=headers,
    722     chunked=chunked,
    723 )
    725 # If we're going to release the connection in ``finally:``, then
    726 # the response doesn't need to know about the connection. Otherwise
    727 # it will also try to release it and we'll have a double-release
    728 # mess.

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connectionpool.py:404, in HTTPConnectionPool._make_request(self, conn, method, url, timeout, chunked, **httplib_request_kw)
    403 try:
--> 404     self._validate_conn(conn)
    405 except (SocketTimeout, BaseSSLError) as e:
    406     # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connectionpool.py:1058, in HTTPSConnectionPool._validate_conn(self, conn)
   1057 if not getattr(conn, "sock", None):  # AppEngine might not have  `.sock`
-> 1058     conn.connect()
   1060 if not conn.is_verified:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connection.py:363, in HTTPSConnection.connect(self)
    361 def connect(self):
    362     # Add certificate verification
--> 363     self.sock = conn = self._new_conn()
    364     hostname = self.host

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/urllib3/connection.py:186, in HTTPConnection._new_conn(self)
    185 except SocketError as e:
--> 186     raise NewConnectionError(
    187         self, "Failed to establish a new connection: %s" % e
    188     )
    190 return conn

NewConnectionError: <botocore.awsrequest.AWSHTTPSConnection object at 0x7f665c7e4220>: Failed to establish a new connection: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

EndpointConnectionError                   Traceback (most recent call last)
Cell In[29], line 4
      2 for qa in df_dict:
      3     print(qa["Question"])
----> 4     response = sentence_window_engine.query(qa["Question"])
      5     print(str(response.response))
      6     sentence_chunks.append(response.response)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/core/base_query_engine.py:30, in BaseQueryEngine.query(self, str_or_query_bundle)
     28 if isinstance(str_or_query_bundle, str):
     29     str_or_query_bundle = QueryBundle(str_or_query_bundle)
---> 30 return self._query(str_or_query_bundle)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/query_engine/retriever_query_engine.py:171, in RetrieverQueryEngine._query(self, query_bundle)
    167 with self.callback_manager.event(
    168     CBEventType.QUERY, payload={EventPayload.QUERY_STR: query_bundle.query_str}
    169 ) as query_event:
    170     nodes = self.retrieve(query_bundle)
--> 171     response = self._response_synthesizer.synthesize(
    172         query=query_bundle,
    173         nodes=nodes,
    174     )
    176     query_event.on_end(payload={EventPayload.RESPONSE: response})
    178 return response

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/response_synthesizers/base.py:146, in BaseSynthesizer.synthesize(self, query, nodes, additional_source_nodes, **response_kwargs)
    141     query = QueryBundle(query_str=query)
    143 with self._callback_manager.event(
    144     CBEventType.SYNTHESIZE, payload={EventPayload.QUERY_STR: query.query_str}
    145 ) as event:
--> 146     response_str = self.get_response(
    147         query_str=query.query_str,
    148         text_chunks=[
    149             n.node.get_content(metadata_mode=MetadataMode.LLM) for n in nodes
    150         ],
    151         **response_kwargs,
    152     )
    154     additional_source_nodes = additional_source_nodes or []
    155     source_nodes = list(nodes) + list(additional_source_nodes)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/response_synthesizers/compact_and_refine.py:38, in CompactAndRefine.get_response(self, query_str, text_chunks, prev_response, **response_kwargs)
     34 # use prompt helper to fix compact text_chunks under the prompt limitation
     35 # TODO: This is a temporary fix - reason it's temporary is that
     36 # the refine template does not account for size of previous answer.
     37 new_texts = self._make_compact_text_chunks(query_str, text_chunks)
---> 38 return super().get_response(
     39     query_str=query_str,
     40     text_chunks=new_texts,
     41     prev_response=prev_response,
     42     **response_kwargs,
     43 )

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/response_synthesizers/refine.py:146, in Refine.get_response(self, query_str, text_chunks, prev_response, **response_kwargs)
    142 for text_chunk in text_chunks:
    143     if prev_response is None:
    144         # if this is the first chunk, and text chunk already
    145         # is an answer, then return it
--> 146         response = self._give_response_single(
    147             query_str, text_chunk, **response_kwargs
    148         )
    149     else:
    150         # refine response if possible
    151         response = self._refine_response_single(
    152             prev_response, query_str, text_chunk, **response_kwargs
    153         )

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/response_synthesizers/refine.py:202, in Refine._give_response_single(self, query_str, text_chunk, **response_kwargs)
    198 if response is None and not self._streaming:
    199     try:
    200         structured_response = cast(
    201             StructuredRefineResponse,
--> 202             program(
    203                 context_str=cur_text_chunk,
    204                 **response_kwargs,
    205             ),
    206         )
    207         query_satisfied = structured_response.query_satisfied
    208         if query_satisfied:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/response_synthesizers/refine.py:64, in DefaultRefineProgram.__call__(self, *args, **kwds)
     62     answer = answer.json()
     63 else:
---> 64     answer = self._llm.predict(
     65         self._prompt,
     66         **kwds,
     67     )
     68 return StructuredRefineResponse(answer=answer, query_satisfied=True)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/llms/llm.py:224, in LLM.predict(self, prompt, **prompt_args)
    222 else:
    223     formatted_prompt = self._get_prompt(prompt, **prompt_args)
--> 224     response = self.complete(formatted_prompt)
    225     output = response.text
    227 return self._parse_output(output)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/llms/base.py:223, in llm_completion_callback.<locals>.wrap.<locals>.wrapped_llm_predict(_self, *args, **kwargs)
    213 with wrapper_logic(_self) as callback_manager:
    214     event_id = callback_manager.on_event_start(
    215         CBEventType.LLM,
    216         payload={
   (...)
    220         },
    221     )
--> 223     f_return_val = f(_self, *args, **kwargs)
    224     if isinstance(f_return_val, Generator):
    225         # intercept the generator and add a callback to the end
    226         def wrapped_gen() -> CompletionResponseGen:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/llms/bedrock.py:189, in Bedrock.complete(self, prompt, **kwargs)
    187 request_body = self._provider.get_request_body(prompt, all_kwargs)
    188 request_body_str = json.dumps(request_body)
--> 189 response = completion_with_retry(
    190     client=self._client,
    191     model=self.model,
    192     request_body=request_body_str,
    193     max_retries=self.max_retries,
    194     **all_kwargs,
    195 )["body"].read()
    196 response = json.loads(response)
    197 return CompletionResponse(
    198     text=self._provider.get_text_from_response(response), raw=response
    199 )

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/llms/bedrock_utils.py:202, in completion_with_retry(client, model, request_body, max_retries, stream, **kwargs)
    197         return client.invoke_model_with_response_stream(
    198             modelId=model, body=request_body
    199         )
    200     return client.invoke_model(modelId=model, body=request_body)
--> 202 return _completion_with_retry(**kwargs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/tenacity/__init__.py:289, in BaseRetrying.wraps.<locals>.wrapped_f(*args, **kw)
    287 @functools.wraps(f)
    288 def wrapped_f(*args: t.Any, **kw: t.Any) -> t.Any:
--> 289     return self(f, *args, **kw)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/tenacity/__init__.py:379, in Retrying.__call__(self, fn, *args, **kwargs)
    377 retry_state = RetryCallState(retry_object=self, fn=fn, args=args, kwargs=kwargs)
    378 while True:
--> 379     do = self.iter(retry_state=retry_state)
    380     if isinstance(do, DoAttempt):
    381         try:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/tenacity/__init__.py:314, in BaseRetrying.iter(self, retry_state)
    312 is_explicit_retry = fut.failed and isinstance(fut.exception(), TryAgain)
    313 if not (is_explicit_retry or self.retry(retry_state)):
--> 314     return fut.result()
    316 if self.after is not None:
    317     self.after(retry_state)

File ~/anaconda3/envs/python3/lib/python3.10/concurrent/futures/_base.py:451, in Future.result(self, timeout)
    449     raise CancelledError()
    450 elif self._state == FINISHED:
--> 451     return self.__get_result()
    453 self._condition.wait(timeout)
    455 if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:

File ~/anaconda3/envs/python3/lib/python3.10/concurrent/futures/_base.py:403, in Future.__get_result(self)
    401 if self._exception:
    402     try:
--> 403         raise self._exception
    404     finally:
    405         # Break a reference cycle with the exception in self._exception
    406         self = None

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/tenacity/__init__.py:382, in Retrying.__call__(self, fn, *args, **kwargs)
    380 if isinstance(do, DoAttempt):
    381     try:
--> 382         result = fn(*args, **kwargs)
    383     except BaseException:  # noqa: B902
    384         retry_state.set_exception(sys.exc_info())  # type: ignore[arg-type]

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/llama_index/llms/bedrock_utils.py:200, in completion_with_retry.<locals>._completion_with_retry(**kwargs)
    196 if stream:
    197     return client.invoke_model_with_response_stream(
    198         modelId=model, body=request_body
    199     )
--> 200 return client.invoke_model(modelId=model, body=request_body)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/client.py:553, in ClientCreator._create_api_method.<locals>._api_call(self, *args, **kwargs)
    549     raise TypeError(
    550         f"{py_operation_name}() only accepts keyword arguments."
    551     )
    552 # The "self" in this scope is referring to the BaseClient.
--> 553 return self._make_api_call(operation_name, kwargs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/client.py:989, in BaseClient._make_api_call(self, operation_name, api_params)
    985     maybe_compress_request(
    986         self.meta.config, request_dict, operation_model
    987     )
    988     apply_request_checksum(request_dict)
--> 989     http, parsed_response = self._make_request(
    990         operation_model, request_dict, request_context
    991     )
    993 self.meta.events.emit(
    994     'after-call.{service_id}.{operation_name}'.format(
    995         service_id=service_id, operation_name=operation_name
   (...)
   1000     context=request_context,
   1001 )
   1003 if http.status_code >= 300:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/client.py:1015, in BaseClient._make_request(self, operation_model, request_dict, request_context)
   1013 def _make_request(self, operation_model, request_dict, request_context):
   1014     try:
-> 1015         return self._endpoint.make_request(operation_model, request_dict)
   1016     except Exception as e:
   1017         self.meta.events.emit(
   1018             'after-call-error.{service_id}.{operation_name}'.format(
   1019                 service_id=self._service_model.service_id.hyphenize(),
   (...)
   1023             context=request_context,
   1024         )

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/endpoint.py:119, in Endpoint.make_request(self, operation_model, request_dict)
    113 def make_request(self, operation_model, request_dict):
    114     logger.debug(
    115         "Making request for %s with params: %s",
    116         operation_model,
    117         request_dict,
    118     )
--> 119     return self._send_request(request_dict, operation_model)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/endpoint.py:202, in Endpoint._send_request(self, request_dict, operation_model)
    198 request = self.create_request(request_dict, operation_model)
    199 success_response, exception = self._get_response(
    200     request, operation_model, context
    201 )
--> 202 while self._needs_retry(
    203     attempts,
    204     operation_model,
    205     request_dict,
    206     success_response,
    207     exception,
    208 ):
    209     attempts += 1
    210     self._update_retries_context(context, attempts, success_response)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/endpoint.py:354, in Endpoint._needs_retry(self, attempts, operation_model, request_dict, response, caught_exception)
    352 service_id = operation_model.service_model.service_id.hyphenize()
    353 event_name = f"needs-retry.{service_id}.{operation_model.name}"
--> 354 responses = self._event_emitter.emit(
    355     event_name,
    356     response=response,
    357     endpoint=self,
    358     operation=operation_model,
    359     attempts=attempts,
    360     caught_exception=caught_exception,
    361     request_dict=request_dict,
    362 )
    363 handler_response = first_non_none_response(responses)
    364 if handler_response is None:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/hooks.py:412, in EventAliaser.emit(self, event_name, **kwargs)
    410 def emit(self, event_name, **kwargs):
    411     aliased_event_name = self._alias_event_name(event_name)
--> 412     return self._emitter.emit(aliased_event_name, **kwargs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/hooks.py:256, in HierarchicalEmitter.emit(self, event_name, **kwargs)
    245 def emit(self, event_name, **kwargs):
    246     """
    247     Emit an event by name with arguments passed as keyword args.
    248 
   (...)
    254              handlers.
    255     """
--> 256     return self._emit(event_name, kwargs)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/hooks.py:239, in HierarchicalEmitter._emit(self, event_name, kwargs, stop_on_response)
    237 for handler in handlers_to_call:
    238     logger.debug('Event %s: calling handler %s', event_name, handler)
--> 239     response = handler(**kwargs)
    240     responses.append((handler, response))
    241     if stop_on_response and response is not None:

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:207, in RetryHandler.__call__(self, attempts, response, caught_exception, **kwargs)
    204     retries_context = kwargs['request_dict']['context'].get('retries')
    205     checker_kwargs.update({'retries_context': retries_context})
--> 207 if self._checker(**checker_kwargs):
    208     result = self._action(attempts=attempts)
    209     logger.debug("Retry needed, action of: %s", result)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:284, in MaxAttemptsDecorator.__call__(self, attempt_number, response, caught_exception, retries_context)
    279 if retries_context:
    280     retries_context['max'] = max(
    281         retries_context.get('max', 0), self._max_attempts
    282     )
--> 284 should_retry = self._should_retry(
    285     attempt_number, response, caught_exception
    286 )
    287 if should_retry:
    288     if attempt_number >= self._max_attempts:
    289         # explicitly set MaxAttemptsReached

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:320, in MaxAttemptsDecorator._should_retry(self, attempt_number, response, caught_exception)
    316         return True
    317 else:
    318     # If we've exceeded the max attempts we just let the exception
    319     # propagate if one has occurred.
--> 320     return self._checker(attempt_number, response, caught_exception)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:363, in MultiChecker.__call__(self, attempt_number, response, caught_exception)
    361 def __call__(self, attempt_number, response, caught_exception):
    362     for checker in self._checkers:
--> 363         checker_response = checker(
    364             attempt_number, response, caught_exception
    365         )
    366         if checker_response:
    367             return checker_response

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:247, in BaseChecker.__call__(self, attempt_number, response, caught_exception)
    245     return self._check_response(attempt_number, response)
    246 elif caught_exception is not None:
--> 247     return self._check_caught_exception(
    248         attempt_number, caught_exception
    249     )
    250 else:
    251     raise ValueError("Both response and caught_exception are None.")

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/retryhandler.py:416, in ExceptionRaiser._check_caught_exception(self, attempt_number, caught_exception)
    408 def _check_caught_exception(self, attempt_number, caught_exception):
    409     # This is implementation specific, but this class is useful by
    410     # coordinating with the MaxAttemptsDecorator.
   (...)
    414     # the MaxAttemptsDecorator is not interested in retrying the exception
    415     # then this exception just propagates out past the retry code.
--> 416     raise caught_exception

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/endpoint.py:281, in Endpoint._do_get_response(self, request, operation_model, context)
    279     http_response = first_non_none_response(responses)
    280     if http_response is None:
--> 281         http_response = self._send(request)
    282 except HTTPClientError as e:
    283     return (None, e)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/endpoint.py:377, in Endpoint._send(self, request)
    376 def _send(self, request):
--> 377     return self.http_session.send(request)

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/botocore/httpsession.py:493, in URLLib3Session.send(self, request)
    491     raise SSLError(endpoint_url=request.url, error=e)
    492 except (NewConnectionError, socket.gaierror) as e:
--> 493     raise EndpointConnectionError(endpoint_url=request.url, error=e)
    494 except ProxyError as e:
    495     raise ProxyConnectionError(
    496         proxy_url=mask_proxy_url(proxy_url), error=e
    497     )
```

### PR DESCRIPTION
**Pull request by sansmoraxz**
# Description

The bedrock models are in broken state with only workaround being langchain. The inconsistency between parameter names is also quite confusing. I am suggesting to reuse the ones provided by boto itself,

Fixes #9812 #9803

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by sansmoraxz** [comment]
The code does need some heavy refactoring for cohere batching. That may be a bit out of scope of this pr aiming for a bugfix.

Will send that separately. Putting it up for review.

File: docs/examples/embeddings/bedrock.ipynb
Comment (logan-markewich) [inline docs/examples/embeddings/bedrock.ipynb:1]: probably we only need to print the first few elements from the embedding vector -- that's a lot of output lol

**Review by logan-markewich — APPROVED** [review]

File: docs/examples/embeddings/bedrock.ipynb
Comment (sansmoraxz) [reply docs/examples/embeddings/bedrock.ipynb:1]: cleared up now.

**Review by logan-markewich — APPROVED** [review]

**Comment by nicoloboschi** [comment]
@sansmoraxz This PR removes some static methods from the class BedrockEmbeddings like `BedrockEmbedding.from_credentials` 

This is a breaking change and I don't get how this is acceptable for the project. 

The description states:
>The bedrock models are in broken state with only workaround being langchain

This is not true, at least for version `0.9.34`. We have running tests with it and the configuration is very simple
```
BedrockEmbedding.from_credentials(
            model_name="amazon.titan-embed-text-v1",
            aws_region="some-region",
        )
```
The credentials are provided by env variables. I think this is a very common setup and it's working with `0.9.34`.

In general, I think this is very bad for dev experience, especially if there are no very strong reason to remove public facing APIs. And, in that case, I would expect something in the release notes. 

My suggestion is to bring back those methods and include them in the next release.  

CC/ @logan-markewich  @jerryjliu

**Comment by sansmoraxz** [comment]
While I agree the changelog should've been updated, this PR was already supposed to be a breaking change.

Maybe the `from_credentials` and `set_credentials` should've been deprecated in a phased manner. But IMO the API was badly designed renaming boto parameters and lacking consistency. AFIK no other embedding class even use that.

And AWS has [credentials provider chain](https://docs.aws.amazon.com/sdkref/latest/guide/standardized-credentials.html), which auto injects required credentials. 
 The og code of redundant checks broken compatibility with Sagemaker or EC2 with IMDS.

**Pull request merged**